### PR TITLE
Monkey patch Lang.get

### PR DESCRIPTION
**Pull request by notbakaneko**
upstream fixes haven't been released yet, either 🚒

### CODE REVIEW

**Comment by nanaya** [comment]
a while ago I wrote this

```coffeescript
Lang.origGet = Lang.get
Lang.get = (key, replacements, locale) ->
  try
    Lang.origGet key, replacements, locale
  catch
    Lang.setLocale fallbackLocale
    message = Lang.origGet key, replacements, locale
    Lang.setLocale currentLocale

    message
```

**Comment by notbakaneko** [comment]
passing in `locale` in the current version of lang.js is broken when it has to do a fallback :D

**Comment by nanaya** [comment]
yeah the thing above "fixes" it.

**Comment by notbakaneko** [comment]
We should probably monkey patch all the calls then ᕕ( ᐛ )ᕗ

File: resources/assets/coffee/lang-js.coffee [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#    warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#    See the GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
+###
+
+# Lang.js monkey patch
+Lang.origGet = Lang.get
+Lang.get = (key, replacements, locale) ->
```
Comment (nanaya) [inline resources/assets/coffee/lang-js.coffee]: Looking again the arguments can probably be removed and call the original method with `.apply null, arguments` instead.

The workaround in `osu.trans` can also be removed.

**Comment by notbakaneko** [comment]
Should probably remove the calls to `osu.trans` too, in a separate pull request

File: resources/assets/coffee/lang-js.coffee [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+###
+
+# Lang.js monkey patch
+Lang.origGet = Lang.get
+Lang.get = ->
+  try
+    Lang.origGet.apply @, arguments
+  catch
+    Lang.setLocale fallbackLocale
+    message = Lang.origGet.apply @, arguments
```
Comment (nanaya) [inline resources/assets/coffee/lang-js.coffee]: apparently this should pass `fallbackLocale` (or nothing at all) instead :D

File: resources/assets/coffee/osu_common.coffee [hunk 3]
```diff
@@ -242,11 +235,6 @@
   transChoice: (key, count, replacements) ->
     message = Lang.choice key, count, replacements, currentLocale
```
Comment (nanaya) [inline resources/assets/coffee/osu_common.coffee]: can be _simplified_.

Comment (notbakaneko) [inline resources/assets/coffee/osu_common.coffee]: That's the original, there's no point changing it since we can just get rid of it

**Comment by peppy** [comment]
good to merge this for now?

**Pull request merged**